naucse/routes.py

```diff
@@ -226,11 +223,14 @@ def session_coverpage(run, session, coverpage):
         rendered session coverpage
     """
 
+    session = run.sessions.get(session)
+
     def session_url(session):
         return url_for('session_coverpage', run=run, session=session, coverpage=coverpage)
 
+    if session.coverpage_available(run.slug, coverpage):
+        content = session.get_coverpage_content(run, coverpage, app)
+    else:
+        content = ""
```

Comment:
I don't see `coverpage_available` useful by itself – usually, you'll call it just before `get_coverpage_content` just to see if you're allowed to make that call. This is known as “Look before you leap” ([LBYL](https://docs.python.org/3.5/glossary.html#term-lbyl)).
Idiomatic Python calls for [EAFP](https://docs.python.org/3.5/glossary.html#term-eafp) instead: just call `get_coverpage_content`, and if it raises an exception, handle it.

Of course `get_coverpage_content` should support this: it should raise a nice exception like `LookupError` if the content is not available. Or just return an empty string, if not having a page is equivalent to that.